spring-integration-amqp/src/main/java/org/springframework/integration/amqp/outbound/AmqpOutboundEndpoint.java

```diff
@@ -111,23 +104,26 @@ public void setExchangeName(String exchangeName) {
 		this.exchangeName = exchangeName;
 	}
 
+	public void setExchangeNameExpression(Expression exchangeNameExpression) {
```

Comment:
This method has been here for a while. Right now it is changed only with the expected type.
Although I won't mind to add JavaDocs...
The `notNull()` isn't necessary. See `doInit()` in this class.